txp/Pos/Txp/Logic/Global.hs

```diff
@@ -45,13 +48,14 @@ txpGlobalSettings =
     }
 
 verifyBlocks
-    :: forall m.
-       TxpGlobalVerifyMode m
+    :: forall ctx m.
+       TxpGlobalVerifyMode ctx m
     => Bool -> OldestFirst NE TxpBlock -> m (OldestFirst NE TxpUndo)
-verifyBlocks verifyAllIsKnown newChain =
-    fst <$> runToilAction @_ @() (mapM verifyDo newChain)
+verifyBlocks verifyAllIsKnown newChain = do
+    epoch <- siEpoch <$> (note ToilUnknownCurEpoch =<< getCurrentSlot)
```

Comment:
> ok. don't we want to reflect in the name of function the fact that only blocks from same epoch can be passed?

Well, I think it would be useful, but it's minor.

> this invariant is true for every block processing function?

Tt is true for every function except all rollback functions and `verifyAndApplyBlocks`.